apps/barista-design-system/src/pages/single-page/components/toc/toc.ts

```diff
@@ -39,46 +35,44 @@ import { DOCUMENT } from '@angular/common';
     class: 'ba-toc',
   },
 })
-export class BaToc implements OnInit, AfterViewInit, OnDestroy {
+export class BaToc implements OnInit {
+  /** Contains the toc items of the page, coming from page-content component */
+  @Input()
+  tocItems: TableOfContents[];
+
   /** @internal all TOC entries */
-  @ViewChildren('headline') _headlines;
+  @ViewChildren('headline') _headlines: QueryList<HTMLElement>;
 
   /** @internal whether the TOC is expanded  */
   _expandToc: boolean;
-  /** @internal all headlines, which should be represented in the TOC  */
-  _headings$ = this._tocService.tocList;
-  /** @internal the TOC entries that are currently active */
-  _activeItems: BaTocItem[] = [];
 
-  /** Subscription on active TOC items */
-  private _activeItemsSubscription = Subscription.EMPTY;
+  /** @internal the TOC items that are currently active */
+  _activeItems: TableOfContents[] = [];
 
   constructor(
-    private _tocService: BaTocService,
-    private _zone: NgZone,
+    private _scrollSpyService: BaScrollSpyService,
     private _platform: Platform,
     @Inject(DOCUMENT) private _document: any,
   ) {}
 
   ngOnInit(): void {
-    this._activeItemsSubscription = this._tocService.activeItems.subscribe(
-      (activeItems) => {
-        this._zone.run(() => {
-          this._activeItems = activeItems;
-        });
-      },
-    );
-  }
-
-  ngAfterViewInit(): void {
-    Promise.resolve().then(() => {
-      const docElement = this._document.getElementById('main') || undefined;
-      this._tocService.genToc(docElement);
-    });
-  }
-
-  ngOnDestroy(): void {
-    this._activeItemsSubscription.unsubscribe();
+    this._scrollSpyService.activeItemId$
+      .asObservable()
+      .subscribe((activeItemId) => {
```

Comment:
please unsubscribe from this subscription!

Reply:
Added unsubscribe by calling `complete()` on the subject in the service when `unspy()` is called. Which is called when destroying the single-page component.